engine/time/src/main/java/io/deephaven/time/calendar/Calendar.java

```diff
@@ -1,280 +1,643 @@
 /**
- * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ * Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
  */
 package io.deephaven.time.calendar;
 
+import io.deephaven.base.verify.Require;
+import io.deephaven.base.verify.RequirementFailure;
 import io.deephaven.time.DateTimeUtils;
 
-import java.time.DayOfWeek;
 import java.time.Instant;
+import java.time.LocalDate;
 import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.deephaven.util.QueryConstants.NULL_INT;
 
 /**
  * A calendar.
- *
- *
- * To comply with the ISO-8601 standard for Dates, Strings should be of the form "yyyy-MM-dd",
- *
- *
- * Methods on Instant may not be precisely defined enough to return an Instant, e.g nextDay(). In these cases, the
- * method will return a String as discussed above.
- *
- *
- * To maintain consistency, each calendar has two fields: a name, and a time zone. A calendar with the same schedule but
- * a different time zone is considered a different calendar.
- *
- *
- * Frequently, the default implementation for methods on Instants is to call the corresponding method on a String with
- * {@code DateTimeUtils.formatDate}. This can be slower than methods written explicitly for DateTimes. If performance is
- * an issue, consider overriding these methods with other behavior.
+ * <p>
+ * A calendar is associated with a specific time zone.
+ * <p>
+ * Date strings must be in a format that can be parsed by {@code DateTimeUtils#parseDate}. Methods that accept strings
+ * can be slower than methods written explicitly for {@code Instant}, {@code ZonedDateTime}, or {@code LocalDate}.
  */
-public interface Calendar {
+public class Calendar {
+
+    private final String name;
+    private final String description;
+    private final ZoneId timeZone;
+
+    // region Constructors
+
+    /**
+     * Creates a new calendar.
+     *
+     * @param name calendar name.
+     * @param description calendar description.
+     * @param timeZone calendar time zone.
+     * @throws RequirementFailure if any parameter is {@code null}
+     */
+    public Calendar(final String name, final String description, final ZoneId timeZone) {
```

Comment:
Is this meant to be user-visible? I see it's only used by test and BusinessCalendar right now. If we need it to be user visible, we should move it to a static method; regardless, we should make this package-private (to support theoretical interface refactor in the future if necessary).